packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte

```diff
@@ -51,26 +43,64 @@
 	let outputJson: string;
 	let text = "";
 
-	async function getOutput({
-		withModelLoading = false,
-		isOnLoadCall = false,
-		exampleOutput = undefined,
-	}: InferenceRunOpts = {}) {
+	let compiledTemplate: Template;
+
+	async function getOutput({ withModelLoading = false, isOnLoadCall = false }: InferenceRunOpts = {}) {
 		const trimmedText = text.trim();
 
 		if (!trimmedText) {
 			return;
 		}
 
-		if (shouldUpdateUrl && !conversation.past_user_inputs.length) {
+		// Error checking
+		const config = model.config;
+		if (config === undefined) {
+			outputJson = "";
+			output = [];
+			error = "Model config not found";
+			return;
+		}
+
+		const tokenizerConfig = config.tokenizer as TokenizerConfig | undefined;
+		if (tokenizerConfig === undefined) {
+			outputJson = "";
+			output = [];
+			error = "Tokenizer config not found";
+			return;
+		}
+
+		if (!compiledTemplate) {
+			const chatTemplate = tokenizerConfig.chat_template;
+			if (chatTemplate === undefined) {
+				outputJson = "";
+				output = [];
+				error = "No chat template found in tokenizer config";
+				return;
+			}
+			compiledTemplate = new Template(chatTemplate);
+		}
```

Comment:
Should be checked `onMount` IMO to avoid the frustrating UX of submitting an input and getting an error back

Reply:
Addressed in https://github.com/huggingface/huggingface.js/pull/457/commits/abf7f8a28bd3d1875ae654b0de7b206413111587